google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -347,7 +348,21 @@ protected void textChanged(DocumentEvent event) {
   private void setManagedSdkUiAvailable(boolean available) {
     if (ServiceManager.getService(PluginInfoService.class).shouldEnable(GctFeature.MANAGED_SDK)) {
       enableAutomaticUpdatesCheckbox.setEnabled(available);
-      checkForUpdatesHyperlink.setVisible(available);
+
+      // only make it visible if managed SDK is active, not currently installing or updating, and
+      // not up-to-date.
+      if (available
+          && CloudSdkServiceUserSettings.getInstance().getUserSelectedSdkServiceType()
+              == CloudSdkServiceType.MANAGED_SDK) {
+        ManagedCloudSdkService managedCloudSdkService =
+            (ManagedCloudSdkService) CloudSdkService.getInstance();
+        if (managedCloudSdkService.getStatus() == SdkStatus.READY
```

Comment:
what happens if this if statement is false? is it correct not to set the visibility of the button?

Reply:
It assumes SDK either not ready or already up to date and doesn't show update button since no update is possible. Do you mean we still want to show it?

Reply:
no saying we should or shouldn't show it, just pointing out that in this branch (where its false) you are neither setting the visibly to true or false.